dace/frontend/common/op_repository.py

```diff
@@ -1,10 +1,30 @@
 # Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
-from typing import Any, Callable, Dict, Tuple
-from dace.dtypes import paramdec
+import itertools
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+from dace.dtypes import paramdec, deduplicate
 
 MethodType = Callable[..., Tuple[str]]
 
 
+def _get_all_bases(class_or_name: Union[str, Type]) -> List[str]:
+    """
+    Returns a list of the current class name and all its base classes.
+
+    :param class_or_name: A class type or a class name.
+    :return: A list of strings representing class names if a type was given, or a list with a single
+             string if a string was given. The list is given in reverse order, with subclasses preceding
+             superclasses.
+    """
+    if isinstance(class_or_name, str):
+        return [class_or_name]
+
+    classes = [class_or_name.__name__]
+    for base in class_or_name.__bases__:
+        classes.extend(_get_all_bases(base))
+
+    return deduplicate(classes)
```

Comment:
This should retain class inheritance order. For example, `View` has `Array` as a base, and the repository should prefer implementations for `View` above implementations for `Array`.

Reply:
Ah, I see now that deduplicate handles this correctly, never mind!